In [1]:
import pandas as pd
import numpy as np
import requests
import json
import tweepy
import os

In [2]:
twitarch_df = pd.read_csv('twitter-archive-enhanced-2.csv')

In [7]:
!conda list

# packages in environment at /home/nemo/miniconda3/envs/doggie:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
argon2-cffi               20.1.0           py38h7b6447c_1  
attrs                     19.3.0                     py_0  
backcall                  0.2.0                      py_0  
blas                      1.0                         mkl  
bleach                    3.1.5                      py_0  
blinker                   1.4                      py38_0  
brotlipy                  0.7.0           py38h7b6447c_1000  
ca-certificates           2020.6.24                     0  
certifi                   2020.6.20                py38_0  
cffi                      1.14.1           py38he30daa8_0  
chardet                   3.0.4                 py38_1003  
cryptography              2.9.2            py38h1ba5d50_0  
cycler                    0.10.0                   py38_0  
dbus                     

In [3]:
# Are all the tweet_ids unique?
print(f"There are {twitarch_df.shape[0]} entries and there are {twitarch_df.tweet_id.nunique()} unique tweet_ids")

There are 2356 entries and there are 2356 unique tweet_ids


All tweet_ids are unique since there are 2356 entries and there are 2356 unique IDs

In [4]:
#twitarch_df.head()

In [5]:
# https://developer.twitter.com/en/docs/authentication/guides/authentication-best-practices
consumer_key = os.environ.get("CONSUMER_KEY")
consumer_secret = os.environ.get("CONSUMER_SECRET")

# from http://docs.tweepy.org/en/latest/getting_started.html
# using Oauth 2
auth = tweepy.AppAuthHandler(consumer_key, consumer_secret)
# https://knowledge.udacity.com/questions/66949#66975
api = tweepy.API(auth_handler = auth,
                 parser = tweepy.parsers.JSONParser(),
                 wait_on_rate_limit = True,
                 wait_on_rate_limit_notify = True)

```
Rate limit reached. Sleeping for: 607
Rate limit reached. Sleeping for: 606
```

### code above ran for 32 minutes
I started execution at 14:30 on 2020-08-24 (Aug 24 2020). It finished around 15:02

In [6]:
# get the info from json object that will be merged with original df
added_features = []

with open('tweet_json.txt', 'r') as file:
    for item in file:
        data = json.loads(item)
        added_features.append({'tweet_id': data['id'],
                               'favorite_count': data['favorite_count'],
                               'retweet_count': data['retweet_count']})

added_fields = pd.DataFrame(added_features)

In [7]:
added_fields.head()

tweet_id  favorite_count  retweet_count
0  892420643555336193           35783           7568
1  892177421306343426           30896           5607
2  891815181378084864           23249           3714
3  891689557279858688           39055           7746
4  891327558926688256           37301           8346